connectors/redis/redis.go

```diff
@@ -0,0 +1,255 @@
+package redis
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/uber-go/dosa"
+	"github.com/uber-go/dosa/connectors/base"
+)
+
+const keySeparator = ","
+
+// SimpleRedis is a minimal interface to Redis commands
+type SimpleRedis interface {
+	Get(key string) ([]byte, error)
+	SetEx(key string, value []byte, ttl time.Duration) error
+	Del(key string) error
+}
+
+// ErrNotImplemented is returned for interface methods that do not have an implementation
+type ErrNotImplemented struct{}
+
+// Error returns a constant string "Not implemented"
+func (*ErrNotImplemented) Error() string {
+	return "Not implemented"
+}
+
+// ErrInvalidEntity is returned for dosa entities that cannot be categorized as a key-value schema for redis
+type ErrInvalidEntity struct {
+	msg string
+}
+
+// Error returns why the schema does not conform to key-value format
+func (e *ErrInvalidEntity) Error() string {
+	return fmt.Sprintf("Invalid entity for Redis. Not key-value schema. %v", e.msg)
+}
+
+// NewErrInvalidEntity returns an ErrInvalidEntity
+func NewErrInvalidEntity(msg string) *ErrInvalidEntity {
+	return &ErrInvalidEntity{msg: msg}
+}
+
+// Config holds the settings for a RedisConnector
+type Config struct {
+	// ServerSettings are the settings specific to redis server
+	ServerSettings ServerConfig
+	// TTL for how long values should live in the cache
+	TTL time.Duration
+}
+
+// ServerConfig holds the settings for redis
+type ServerConfig struct {
+	Host string
+	Port int
+	// MaxIdle is the maximum number of idle connections in the pool.
+	MaxIdle int
+	// IdleTimeout directs to close connections after remaining idle for this duration.
+	// If the value is zero, then idle connections are not closed. Applications should set
+	// the timeout to a value less than the server's timeout.
+	IdleTimeout time.Duration
+	// Maximum number of connections allocated by the pool at a given time.
+	// When zero, there is no limit on the number of connections in the pool.
+	MaxActive int
+
+	ConnectTimeout time.Duration
+	ReadTimeout    time.Duration
+	WriteTimeout   time.Duration
+}
+
+// NewConnector initializes a Redis Connector
+func NewConnector(config Config) dosa.Connector {
+	return &Connector{
+		client: NewRedigoClient(config.ServerSettings),
+		ttl:    config.TTL,
+	}
+}
+
+// Connector for redis database
+type Connector struct {
+	base.Connector
+	client SimpleRedis
+	ttl    time.Duration
+}
+
+// CreateIfNotExists not implemented
+func (c *Connector) CreateIfNotExists(ctx context.Context, ei *dosa.EntityInfo, values map[string]dosa.FieldValue) error {
+	return new(ErrNotImplemented)
```

Comment:
Although not needed, this is pretty easy to implement, except that it requires a change to the SimpleRedis interface